Redesign Compression Bad Chunks Tests

stwrt opened the pull request:
We cannot guarantee that the stream will error until we close the stream.
Therefore, we shall first write the chunks and do a read ignoring any possible errors.
Next, we shall close the writable stream, and now start checking for errors.
Finally, do a follow up read to ensure that the read is throwing errors too.

ricea commented on compression/compression-bad-chunks.tentative.any.js:
```diff
@@ -39,36 +39,67 @@ const badChunks = [
-    const readPromise = reader.read();
-    await promise_rejects_js(t, TypeError, writePromise, 'write should reject');
-    await promise_rejects_js(t, TypeError, readPromise, 'read should reject');
+    const ds = new DecompressionStream('gzip');
```
I'm confused. It looks like you're changing the compression tests to be decompression tests?

stwrt replied:
You are right. I messed up here and copied my changes to the wrong file.

Thanks for catching this. Will have a new patch up this evening with the fix.

stwrt replied:
Uploaded new patch with the correct file modified.

stwrt committed:
Redesign Decompression Bad Chunks Tests

We cannot guarantee that the stream will error until we close the stream.
Therefore, we shall first write the chunks and do a read ignoring any possible errors.
Next, we shall close the writable stream, and now start checking for errors.
Finally, do a follow up read to ensure that the read is throwing errors too.

youennf reviewed (approved):
LGTM.
A further refactoring could be to share more code between the 3 tests.
It seems the code is the same for all 3 tests, except for the DecompressionStream constructor parameter.

youennf commented on compression/decompression-bad-chunks.tentative.any.js:
```diff
@@ -49,34 +49,57 @@ const badChunks = [
+    await promise_rejects_js(t, TypeError, writer.closed, 'write should reject');
+
+    await promise_rejects_js(t, TypeError, reader.read(), 'read should reject');
+    await promise_rejects_js(t, TypeError, reader.closed, 'read should reject');
```
Ditto here 'reader read should reject' and 'reader closed should reject'.
Same for the two other places

stwrt commented:
> LGTM.
> A further refactoring could be to share more code between the 3 tests.
> It seems the code is the same for all 3 tests, except for the DecompressionStream constructor parameter.

Refactored the code for all decompression to be handled in a single function.
